src/rust/bin/inspect/transaction.rs

```diff
@@ -0,0 +1,555 @@
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+};
+
+use bellman::groth16;
+use group::GroupEncoding;
+use orchard::note_encryption::OrchardDomain;
+use secp256k1::{Secp256k1, VerifyOnly};
+use zcash_address::{
+    unified::{self, Encoding},
+    ToAddress, ZcashAddress,
+};
+use zcash_note_encryption::try_output_recovery_with_ovk;
+#[allow(deprecated)]
+use zcash_primitives::{
+    consensus::BlockHeight,
+    legacy::{keys::pubkey_to_address, Script, TransparentAddress},
+    memo::{Memo, MemoBytes},
+    sapling::note_encryption::SaplingDomain,
+    transaction::{
+        components::{orchard as orchard_serialization, sapling, transparent, Amount},
+        sighash::{signature_hash, SignableInput, TransparentAuthorizingContext},
+        txid::TxIdDigester,
+        Authorization, Transaction, TransactionData, TxVersion,
+    },
+};
+use zcash_proofs::sapling::SaplingVerificationContext;
+
+use crate::{context::Context, GROTH16_PARAMS, ORCHARD_VK};
+
+pub fn is_coinbase(tx: &Transaction) -> bool {
+    tx.transparent_bundle()
+        .map(|b| b.is_coinbase())
+        .unwrap_or(false)
+}
+
+pub fn extract_height_from_coinbase(tx: &Transaction) -> Option<BlockHeight> {
```

Comment:
These also seem like they'd be useful library functions. Or perhaps even methods on `Transaction`?